src/InputField/index.js

```diff
@@ -323,29 +323,93 @@ const InputField = React.forwardRef((props: Props, ref: Ref) => {
     inputMode,
   } = props;
 
-  const forID = id || (label ? React.useMemo(() => randomID("inputFieldID"), []) : undefined);
+  const forID = React.useMemo(() => {
+    if (id) return id;
+    return randomID("inputFieldID");
+  }, [id]);
+  const [tooltipShown, setTooltipShown] = useState(false);
+  const [tooltipShownHover, setTooltipShownHover] = useState(false);
+  const labelRef = useRef(null);
+  const iconRef = useRef(null);
+
+  const handleFocus = useCallback(
```

Comment:
Don't you wanna use the `useStateWithCallback` hook? It would save a lot of LOC. You would only memoized (for now 😉) the passed callback.
```jsx
const callback = useCallback(ev => state => {
 if(onFocus && state) onFocus(ev);
 if(onBlur && !state) onBlur(ev);
}, [onFocus, onBlur]);
const [tooltipShown, setTooltipShown] = useStateWithCallback(false, callback);
```

Or it could be custom hook as we talked yesterday.